## 1. The oldest businesses in the world
<p>This is Staffelter Hof Winery, Germany's oldest business, which was established in 862 under the Carolingian dynasty. It has continued to serve customers through dramatic changes in Europe such as the Holy Roman Empire, the Ottoman Empire, and both world wars. What characteristics enable a business to stand the test of time? Image credit: <a href="https://commons.wikimedia.org/wiki/File:MKn_Staffelter_Hof.jpg">Martin Kraft</a>
<img src="https://assets.datacamp.com/production/project_1383/./img/MKn_Staffelter_Hof.jpeg" alt="The entrance to Staffelter Hof Winery, a German winery established in 862." width="400"></p>
<p>To help answer this question, BusinessFinancing.co.uk <a href="https://businessfinancing.co.uk/the-oldest-company-in-almost-every-country">researched</a> the oldest company that is still in business in almost every country and compiled the results into a dataset. Let's explore this work to to better understand these historic businesses. Our datasets, which are all located in the <code>datasets</code> directory, contain the following information: </p>
<h3 id="businessesandnew_businesses"><code>businesses</code> and <code>new_businesses</code></h3>
<table>
<thead>
<tr>
<th style="text-align:left;">column</th>
<th>type</th>
<th>meaning</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;"><code>business</code></td>
<td>varchar</td>
<td>Name of the business.</td>
</tr>
<tr>
<td style="text-align:left;"><code>year_founded</code></td>
<td>int</td>
<td>Year the business was founded.</td>
</tr>
<tr>
<td style="text-align:left;"><code>category_code</code></td>
<td>varchar</td>
<td>Code for the category of the business.</td>
</tr>
<tr>
<td style="text-align:left;"><code>country_code</code></td>
<td>char</td>
<td>ISO 3166-1 3-letter country code.</td>
</tr>
</tbody>
</table>
<h3 id="countries"><code>countries</code></h3>
<table>
<thead>
<tr>
<th style="text-align:left;">column</th>
<th>type</th>
<th>meaning</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;"><code>country_code</code></td>
<td>varchar</td>
<td>ISO 3166-1 3-letter country code.</td>
</tr>
<tr>
<td style="text-align:left;"><code>country</code></td>
<td>varchar</td>
<td>Name of the country.</td>
</tr>
<tr>
<td style="text-align:left;"><code>continent</code></td>
<td>varchar</td>
<td>Name of the continent that the country exists in.</td>
</tr>
</tbody>
</table>
<h3 id="categories"><code>categories</code></h3>
<table>
<thead>
<tr>
<th style="text-align:left;">column</th>
<th>type</th>
<th>meaning</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;"><code>category_code</code></td>
<td>varchar</td>
<td>Code for the category of the business.</td>
</tr>
<tr>
<td style="text-align:left;"><code>category</code></td>
<td>varchar</td>
<td>Description of the business category.</td>
</tr>
</tbody>
</table>
<p>Now let's learn about some of the world's oldest businesses still in operation!</p>

In [4]:
# Import the pandas library under its usual alias 
import pandas as pd

# Load the business.csv file as a DataFrame called businesses
businesses = pd.read_csv("datasets/What and Where are the World's Oldest Businesses/businesses.TXT")
print(businesses)

# Sort businesses from oldest businesses to youngest
sorted_businesses = businesses.sort_values('year_founded', ascending = True)

# Display the first few lines of sorted_businesses
sorted_businesses.head()


                           business  year_founded category_code country_code
0                    Hamoud Boualem          1878         CAT11          DZA
1    Communauté Électrique du Bénin          1968         CAT10          BEN
2          Botswana Meat Commission          1965          CAT1          BWA
3                       Air Burkina          1967          CAT2          BFA
4                           Brarudi          1955          CAT9          BDI
..                              ...           ...           ...          ...
158                  Cafe Brasilero          1877          CAT4          URY
159                  Hacienda Chuao          1660         CAT11          VEN
160                  Australia Post          1809         CAT16          AUS
161             Bank of New Zealand          1861          CAT3          NZL
162          European Trust Company          1991          CAT3          VUT

[163 rows x 4 columns]


business  year_founded category_code country_code
64                    Kongō Gumi           578          CAT6          JPN
94   St. Peter Stifts Kulinarium           803          CAT4          AUT
107        Staffelter Hof Winery           862          CAT9          DEU
106            Monnaie de Paris            864         CAT12          FRA
103               The Royal Mint           886         CAT12          GBR

In [5]:
categories = pd.read_csv("datasets/What and Where are the World's Oldest Businesses/categories.TXT")
categories

category_code                           category
0           CAT1                        Agriculture
1           CAT2               Aviation & Transport
2           CAT3                  Banking & Finance
3           CAT4          Cafés, Restaurants & Bars
4           CAT5                       Conglomerate
5           CAT6                       Construction
6           CAT7                     Consumer Goods
7           CAT8                            Defense
8           CAT9  Distillers, Vintners, & Breweries
9          CAT10                             Energy
10         CAT11                   Food & Beverages
11         CAT12         Manufacturing & Production
12         CAT13                              Media
13         CAT14                            Medical
14         CAT15                             Mining
15         CAT16                     Postal Service
16         CAT17                             Retail
17         CAT18                 Telecommunications
18         CAT19                   Tourism & Hotels

In [6]:
countries = pd.read_csv("datasets/What and Where are the World's Oldest Businesses/countries.TXT")
countries

country_code               country continent
0            AFG           Afghanistan      Asia
1            AGO                Angola    Africa
2            ALB               Albania    Europe
3            AND               Andorra    Europe
4            ARE  United Arab Emirates      Asia
..           ...                   ...       ...
190          YEM                 Yemen      Asia
191          ZAF          South Africa    Africa
192          ZMB                Zambia    Africa
193          ZWE              Zimbabwe    Africa
194           XK                Kosovo    Europe

[195 rows x 3 columns]

In [7]:
new_businesses = pd.read_csv("datasets/What and Where are the World's Oldest Businesses/new_businesses.TXT")
new_businesses

business  year_founded category_code country_code
0                 Fiji Times          1869         CAT13          FJI
1  J. Armando Bermúdez & Co.          1852          CAT9          DOM

## 2. The oldest businesses in North America
<p>So far we've learned that Kongō Gumi is the world's oldest continuously operating business, beating out the second oldest business by well over 100 years! It's a little hard to read the country codes, though. Wouldn't it be nice if we had a list of country names to go along with the country codes?</p>
<p>Enter <code>countries.csv</code>, which is also located in the <code>datasets</code> folder. Having useful information in different files is a common problem: for data storage, it's better to keep different types of data separate, but for analysis, we want all the data in one place. To solve this, we'll have to join the two tables together. </p>
<h3 id="countries"><code>countries</code></h3>
<table>
<thead>
<tr>
<th style="text-align:left;">column</th>
<th>type</th>
<th>meaning</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;"><code>country_code</code></td>
<td>varchar</td>
<td>ISO 3166-1 3-letter country code.</td>
</tr>
<tr>
<td style="text-align:left;"><code>country</code></td>
<td>varchar</td>
<td>Name of the country.</td>
</tr>
<tr>
<td style="text-align:left;"><code>continent</code></td>
<td>varchar</td>
<td>Name of the continent that the country exists in.</td>
</tr>
</tbody>
</table>
<p>Since <code>countries.csv</code> contains a <code>continent</code> column, merging the datasets will also allow us to look at the oldest business on each continent! </p>

In [8]:
# Load countries.csv to a DataFrame
countries = pd.read_csv("datasets/What and Where are the World's Oldest Businesses/countries.TXT")

# Merge sorted_businesses with countries
businesses_countries = sorted_businesses.merge(countries, on = 'country_code')

# Filter businesses_countries to include countries in North America only
north_america = businesses_countries[businesses_countries.continent == 'North America']
north_america.head()

business  year_founded category_code country_code  \
22  La Casa de Moneda de México          1534         CAT12          MEX   
28           Shirley Plantation          1638          CAT1          USA   
33         Hudson's Bay Company          1670         CAT17          CAN   
35                Mount Gay Rum          1703          CAT9          BRB   
40                    Rose Hall          1770         CAT19          JAM   

          country      continent  
22         Mexico  North America  
28  United States  North America  
33         Canada  North America  
35       Barbados  North America  
40        Jamaica  North America

## 3. The oldest business on each continent
<p>Now we can see that the oldest company in North America is La Casa de Moneda de México, founded in 1534. Why stop there, though, when we could easily find out the oldest business on every continent? </p>

In [9]:
# Create continent, which lists only the continent and oldest year_founded
continent = businesses_countries.groupby("continent").agg({"year_founded" : "min"})

# Merge continent with businesses_countries
merged_continent = continent.merge(businesses_countries, on = 'year_founded')

# Subset continent so that only the four columns of interest are included
subset_merged_continent = merged_continent[["continent","country","business","year_founded"]]
subset_merged_continent

continent    country                     business  year_founded
0         Africa  Mauritius               Mauritius Post          1772
1           Asia      Japan                   Kongō Gumi           578
2         Europe    Austria  St. Peter Stifts Kulinarium           803
3  North America     Mexico  La Casa de Moneda de México          1534
4        Oceania  Australia               Australia Post          1809
5  South America       Peru      Casa Nacional de Moneda          1565

## 4. Unknown oldest businesses
<p>BusinessFinancing.co.uk wasn't able to determine the oldest business for some countries, and those countries are simply left off of <code>businesses.csv</code> and, by extension, <code>businesses</code>. However, the <code>countries</code> that we created <em>does</em> include all countries in the world, regardless of whether the oldest business is known. </p>
<p>We can compare the two datasets in one DataFrame to find out which countries don't have a known oldest business! </p>

In [10]:
# Use .merge() to create a DataFrame, all_countries
all_countries = businesses.merge(countries, on ='country_code', how = 'right', indicator = True)

# Filter to include only countries without oldest businesses
missing_countries = all_countries[all_countries["_merge"] != "both"]

# Create a series of the country names with missing oldest business data
missing_countries_series = missing_countries["country"]

# Display the series
missing_countries_series

163                              Angola
164                 Antigua and Barbuda
165                             Bahamas
166                  Dominican Republic
167                             Ecuador
168                                Fiji
169     Micronesia, Federated States of
170                               Ghana
171                              Gambia
172                             Grenada
173           Iran, Islamic Republic of
174                          Kyrgyzstan
175                            Kiribati
176               Saint Kitts and Nevis
177                              Monaco
178                Moldova, Republic of
179                            Maldives
180                    Marshall Islands
181                               Nauru
182                               Palau
183                    Papua New Guinea
184                            Paraguay
185                 Palestine, State of
186                     Solomon Islands
187                            Suriname


## 5. Adding new oldest business data
<p>It looks like we've got some holes in our dataset! Fortunately, we've taken it upon ourselves to improve upon BusinessFinancing.co.uk's work and find oldest businesses in a few of the missing countries. We've stored the newfound oldest businesses in <code>new_businesses</code>, located at <code>"datasets/new_businesses.csv"</code>. It has the exact same structure as our <code>businesses</code> dataset. </p>
<h3 id="new_businesses"><code>new_businesses</code></h3>
<table>
<thead>
<tr>
<th style="text-align:left;">column</th>
<th>type</th>
<th>meaning</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;"><code>business</code></td>
<td>varchar</td>
<td>Name of the business.</td>
</tr>
<tr>
<td style="text-align:left;"><code>year_founded</code></td>
<td>int</td>
<td>Year the business was founded.</td>
</tr>
<tr>
<td style="text-align:left;"><code>category_code</code></td>
<td>varchar</td>
<td>Code for the category of the business.</td>
</tr>
<tr>
<td style="text-align:left;"><code>country_code</code></td>
<td>char</td>
<td>ISO 3166-1 3-letter country code.</td>
</tr>
</tbody>
</table>
<p>All we have to do is combine the two so that we've got one more complete list of businesses!</p>

In [11]:
# Import new_businesses.csv
new_businesses = pd.read_csv("datasets/What and Where are the World's Oldest Businesses/new_businesses.TXT")

# Add the data in new_businesses to the existing businesses
all_businesses = pd.concat([new_businesses,businesses])

# Merge and filter to find countries with missing business data
new_all_countries = all_businesses.merge(countries, on = 'country_code', indicator = True, how = 'right')
new_missing_countries = new_all_countries[new_all_countries['_merge']!='both']

# Group by continent and create a "count_missing" column
count_missing = new_missing_countries.groupby('continent').agg({'country':'count'})

#count_missing.columns = [...]
count_missing = count_missing.rename(columns = {'country': 'count_missing'})
count_missing


count_missing
continent                   
Africa                     3
Asia                       7
Europe                     2
North America              5
Oceania                   10
South America              3

## 6. The oldest industries
<p>Remember our oldest business in the world, Kongō Gumi? </p>
<table>
<thead>
<tr>
<th style="text-align:right;"></th>
<th style="text-align:left;">business</th>
<th style="text-align:right;">year_founded</th>
<th style="text-align:left;">category_code</th>
<th style="text-align:left;">country_code</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">64</td>
<td style="text-align:left;">Kongō Gumi</td>
<td style="text-align:right;">578</td>
<td style="text-align:left;">CAT6</td>
<td style="text-align:left;">JPN</td>
</tr>
</tbody>
</table>
<p>We know Kongō Gumi was founded in the year 578 in Japan, but it's a little hard to decipher which industry it's in. Information about what the <code>category_code</code> column refers to is in <code>"datasets/categories.csv"</code>: </p>
<h3 id="categories"><code>categories</code></h3>
<table>
<thead>
<tr>
<th style="text-align:left;">column</th>
<th>type</th>
<th>meaning</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;"><code>category_code</code></td>
<td>varchar</td>
<td>Code for the category of the business.</td>
</tr>
<tr>
<td style="text-align:left;"><code>category</code></td>
<td>varchar</td>
<td>Description of the business category.</td>
</tr>
</tbody>
</table>
<p>Let's use <code>categories.csv</code> to understand how many oldest businesses are in each category of industry.</p>

In [12]:
# Import categories.csv and merge to businesses
categories = pd.read_csv("datasets/What and Where are the World's Oldest Businesses/categories.TXT")
businesses_categories = businesses.merge(categories, on = 'category_code')

# Create a DataFrame which lists the number of oldest businesses in each category
count_business_cats = businesses_categories.groupby('category').agg({'business':'count'})

# Create a DataFrame which lists the cumulative years businesses from each category have been operating
years_business_cats = businesses_categories.groupby("category").agg({'year_founded':'sum'})

# Rename columns and display the first five rows of both DataFrames
count_business_cats.columns = ['count']
years_business_cats.columns = ['total_years_in_business']
display(count_business_cats.head(), years_business_cats.head())

count
category                        
Agriculture                    6
Aviation & Transport          19
Banking & Finance             37
Cafés, Restaurants & Bars      6
Conglomerate                   3

total_years_in_business
category                                          
Agriculture                                  10669
Aviation & Transport                         36598
Banking & Finance                            70302
Cafés, Restaurants & Bars                     8532
Conglomerate                                  5671

## 7. Restaurant representation
<p>No matter how we measure it, looks like Banking and Finance is an excellent industry to be in if longevity is our goal! Let's zoom in on another industry: cafés, restaurants, and bars. Which restaurants in our dataset have been around since before the year 1800?</p>

In [13]:
# Filter using .query() for CAT4 businesses founded before 1800; sort results
old_restaurants = businesses_categories \
                 .query('category_code == "CAT4" and year_founded < 1800')

# Sort the DataFrame
old_restaurants = old_restaurants.sort_values(by = 'year_founded', ascending = True)
old_restaurants

business  year_founded category_code  \
142         St. Peter Stifts Kulinarium           803          CAT4   
143                          Sean's Bar           900          CAT4   
139  Ma Yu Ching's Bucket Chicken House          1153          CAT4   

    country_code                   category  
142          AUT  Cafés, Restaurants & Bars  
143          IRL  Cafés, Restaurants & Bars  
139          CHN  Cafés, Restaurants & Bars

## 8. Categories and continents
<p>St. Peter Stifts Kulinarium is old enough that the restaurant is believed to have served Mozart - and it would have been over 900 years old even when he was a patron! Let's finish by looking at the oldest business in each category of commerce for each continent. </p>

In [14]:
# Merge all businesses, countries, and categories together
businesses_categories_countries = pd.merge(businesses, categories, on = 'category_code') \
                .merge(countries, on = 'country_code')

# Sort businesses_categories_countries from oldest to most recent
businesses_categories_countries = businesses_categories_countries.sort_values(by = 'year_founded', ascending = True)

# Create the oldest by continent and category DataFrame
oldest_by_continent_category = businesses_categories_countries.groupby(["continent","category"]).agg({'year_founded':'min'})
oldest_by_continent_category

year_founded
continent     category                                       
Africa        Agriculture                                1947
              Aviation & Transport                       1854
              Banking & Finance                          1892
              Distillers, Vintners, & Breweries          1933
              Energy                                     1968
              Food & Beverages                           1878
              Manufacturing & Production                 1820
              Media                                      1943
              Mining                                     1962
              Postal Service                             1772
Asia          Agriculture                                1930
              Aviation & Transport                       1858
              Banking & Finance                          1830
              Cafés, Restaurants & Bars                  1153
              Conglomerate                               1841
              Construction                                578
              Defense                                    1808
              Distillers, Vintners, & Breweries          1853
              Energy                                     1928
              Food & Beverages                           1820
              Manufacturing & Production                 1736
              Media                                      1931
              Mining                                     1913
              Postal Service                             1800
              Retail                                     1883
              Telecommunications                         1885
              Tourism & Hotels                           1584
Europe        Agriculture                                1218
              Banking & Finance                          1606
              Cafés, Restaurants & Bars                   803
              Consumer Goods                             1649
              Defense                                    1878
              Distillers, Vintners, & Breweries           862
              Manufacturing & Production                  864
              Media                                      1999
              Medical                                    1422
              Mining                                     1248
              Postal Service                             1520
              Telecommunications                         1912
              Tourism & Hotels                           1230
North America Agriculture                                1638
              Aviation & Transport                       1870
              Banking & Finance                          1891
              Distillers, Vintners, & Breweries          1703
              Food & Beverages                           1920
              Manufacturing & Production                 1534
              Media                                      1909
              Retail                                     1670
              Tourism & Hotels                           1770
Oceania       Banking & Finance                          1861
              Postal Service                             1809
South America Banking & Finance                          1565
              Cafés, Restaurants & Bars                  1877
              Defense                                    1811
              Food & Beverages                           1660
              Manufacturing & Production                 1621